### PR TITLE
308 add sqlalchemy 20 support

### PR DESCRIPTION
**Pull request by elisallenens**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,10 +51,10 @@ jobs:
             pins: "h5py==3.7.* sqlalchemy==1.4.40 shapely==2.0.* pyproj==3.4.*"
             extras: "[test,gridadmin]"
           # current
-          - python: '3.10'  # h5py has no wheels for 3.11 (yet) https://github.com/h5py/h5py/issues/2146
+          - python: '3.11'
             os: ubuntu-latest
             numpy: numpy
-            pins: "git+https://github.com/nens/threedi-modelchecker@master#egg=threedi-modelchecker"
+            pins: "h5py>=3.8.* sqlalchemy>=2.0.* shapely==2.0.* pyproj==3.4.*"
```
Comment (caspervdw) [inline .github/workflows/test.yml:57]: Please don't add version pins here; I want to test with the latest ones in the 'current'

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -84,7 +84,7 @@ jobs:
         run: |
           pip install --disable-pip-version-check --upgrade pip setuptools==63.* wheel scikit-build 
           pip install ${{ matrix.numpy }}
-          pip install -e .${{ matrix.extras }} --no-build-isolation ${{ matrix.pins }} pygdal==$(gdal-config --version).*
+          pip install -e .${{ matrix.extras }} --no-build-isolation ${{ matrix.pins }} "pygdal>=$(gdal-config --version).7,==$(gdal-config --version).*"
```
Comment (caspervdw) [inline .github/workflows/test.yml:87]:
```suggestion
          pip install -e .${{ matrix.extras }} --no-build-isolation ${{ matrix.pins }} "pygdal==$(gdal-config --version).*"
```

**Review by caspervdw — COMMENTED** [review]
Good work! two minor thing things in the Actions file to cleanup

File: .github/workflows/test.yml (hunk 2 above)
Comment (elisallenens) [reply .github/workflows/test.yml:87]: that failed in https://github.com/nens/threedigrid-builder/actions/runs/4196155800

**Pull request merged**